snapcraft/internal/parser.py

```diff
@@ -191,12 +191,11 @@ def _process_entry(data):
     origin_dir = os.path.join(_get_base_dir(), _encode_origin(origin))
     os.makedirs(origin_dir, exist_ok=True)
 
-    class Options:
-        source = origin
-        source_type = origin_type
-
-    options = Options()
-    sources.get(origin_dir, None, options)
+    source_handler = sources.get_source_handler(origin,
+                                                source_type=origin_type)
+    handler = source_handler(origin, source_dir=origin_dir)
+    repo.check_for_command(handler.command)
```

Comment:
This doesn't quite feel like the right place to do this. Shouldn't `handler.pull()` be the one checking that it can be called instead of putting that responsibility on the caller?